### PR TITLE
Decrease complexity of code in scout/adapter/mongo/variant to silence codefactor complaints

### PR DESCRIPTION
**Pull request by northwestwitch**
Fixes (partially) #2984 

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by GitHub actions

### CODE REVIEW

File: scout/adapter/mongo/variant.py [hunk 1]
```diff
@@ -832,6 +832,27 @@ def evaluated_variants(self, case_id, institute_id):
...
+        Accepts:
+            case_obj(dict): a case dictionary
+            category(str): "snv", "sv", "str", "cancer"
+            variant_type(str): "clinical" or "research"
+
+        Returns:
+            variant_file(str): string path to a VCF file on disk
+        """
+        variant_file = None
+        for var_type in ["snv", "sv", "str", "research"]:
```
Comment (dnil) [inline scout/adapter/mongo/variant.py:847]:
```suggestion
        for var_type in ["snv", "sv", "str", "cancer"]:
```

File: scout/adapter/mongo/variant.py [hunk 2]
```diff
@@ -832,6 +832,27 @@ def evaluated_variants(self, case_id, institute_id):
         # Return a list with the variant objects
         return variants.values()
 
+    def get_variant_file(self, case_obj, category, variant_type):
+        """Retrieve that should be parsed to extract variants
```
Comment (dnil) [inline scout/adapter/mongo/variant.py:836]:
```suggestion
        """Retrieve file name for file that should be parsed to extract variants
```

File: scout/adapter/mongo/variant.py [hunk 3]
```diff
@@ -849,7 +870,7 @@ def get_region_vcf(
         Args:
             case_obj(dict): A case from the scout database
             variant_type(str): 'clinical' or 'research'. Default: 'clinical'
-            category(str): 'snv' or 'sv'. Default: 'snv'
+            category(str): "snv", "sv", "str", "cancer". Default: 'snv'
```
Comment (dnil) [inline scout/adapter/mongo/variant.py:873]: 👍

File: scout/adapter/mongo/variant.py [hunk 4]
```diff
@@ -861,22 +882,7 @@ def get_region_vcf(
...
-                variant_file = case_obj["vcf_files"].get("vcf_str")
-            elif category == "cancer":
-                variant_file = case_obj["vcf_files"].get("vcf_cancer")
-        elif variant_type == "research":
-            if category == "snv":
-                variant_file = case_obj["vcf_files"].get("vcf_snv_research")
-            elif category == "sv":
-                variant_file = case_obj["vcf_files"].get("vcf_sv_research")
-
+        variant_file = self.get_variant_file(case_obj, category, variant_type)
```
Comment (dnil) [inline scout/adapter/mongo/variant.py:885]: 👍

File: scout/adapter/mongo/variant.py (hunk 1 above)
Comment (dnil) [reply scout/adapter/mongo/variant.py:847]: I dunno, I might have chosen a different temp variable name for the category match as well; var_type to me sounds more like what would be matched to variant_type. But everything works! 😊

**Review by dnil — APPROVED** [review]
That research->cancer typo needs to be fixed, otherwise it will be really nice to be rid of the complexity warning! 🥇

File: scout/adapter/mongo/variant.py (hunk 2 above)
Comment (northwestwitch) [reply scout/adapter/mongo/variant.py:836]: Yes! 😆

File: scout/adapter/mongo/variant.py (hunk 1 above)
Comment (northwestwitch) [reply scout/adapter/mongo/variant.py:847]: We can change it, what do you propose?

Comment (northwestwitch) [reply scout/adapter/mongo/variant.py]: Ah yes, nasty typo!

**Comment by northwestwitch** [comment]
> That research->cancer typo needs to be fixed, otherwise it will be really nice to be rid of the complexity warning! 🥇

Done, thanks! Nice that you noticed it!

**Pull request merged**